### PR TITLE
Migrate Manta Remote state to be a backend

### PR DESCRIPTION
**Pull request by stack72**
This is a WIP to show the progress of this work

### CODE REVIEW

File: backend/init/init.go [hunk 1]
```diff
@@ -47,6 +48,7 @@ func init() {
 			`Warning: "azure" name is deprecated, please use "azurerm"`),
 		"azurerm": func() backend.Backend { return backendAzure.New() },
 		"etcdv3":  func() backend.Backend { return backendetcdv3.New() },
+		"manta":   func() backend.Backend { return backendManta.New() },
```
Comment (cheapRoc) [inline backend/init/init.go]: For historical reasons this was called Manta but arguably that's an internal name for the project now. From the product side of things its Triton Object Store (hence the reason `github.com/joyent/triton-go/storage`). I'm wondering if we should rename this to Triton and deprecate the Manta brand from TF completely.

Thoughts?

Comment (stack72) [reply backend/init/init.go]: Just to make it known, @cheapRoc and I discussed this internally and decided to leave it as manta :)

**Comment by stack72** [comment]
@jbardin would love your thoughts on this please?

**Comment by jbardin** [comment]
Thanks @stack72, I'll take a look at it ASAP. I'd like to get this and the gcs backend conversion both in before the next beta.

File: backend/remote-state/manta/client.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+		info.ID = lockID
+	}
+
+	data := info.Marshal()
+
+	contentType := "application/json"
+	contentLength := int64(len(data))
+
+	params := &storage.PutObjectInput{
```
[the remark this answers is not in the document]
Comment (sean-) [reply backend/remote-state/manta/client.go]: Manta is a strongly consistent backend so this is correct.

**Comment by stack72** [comment]
@jbardin so I took the liberty to remove anything to do with `env:` - it shouldn't even have been in the PR in the first place

I am working on the tests right now

**Comment by stack72** [comment]
@jbardin finally done :)

```
terraform [manta-backend●] % acctests backend/remote-state/manta 2>~/testing-output.txt
=== RUN   TestBackend_impl
--- PASS: TestBackend_impl (0.00s)
=== RUN   TestBackend
--- PASS: TestBackend (41.38s)
=== RUN   TestBackendLocked
--- PASS: TestBackendLocked (49.91s)
	backend_test.go:63: TestBackend: testing state locking for *manta.Backend
=== RUN   TestRemoteClient_impl
--- PASS: TestRemoteClient_impl (0.00s)
=== RUN   TestRemoteClient
--- PASS: TestRemoteClient (5.10s)
=== RUN   TestRemoteClientLocks
--- PASS: TestRemoteClientLocks (10.71s)
PASS
ok  	github.com/hashicorp/terraform/backend/remote-state/manta	107.115s
```

**Comment by stack72** [comment]
Thanks for the merge here @jbardin - glad this will be part of 0.11 GA :)

**Pull request merged**